backend/app/services/database.py

```
@@ -199,385 +199,357 @@ def _initialize_database(self):
-            user_data = cursor.fetchone()
-            if user_data:
-                return dict(user_data)  # sqlite3.Row can be directly converted to dict
-            return None
-        except sqlite3.Error as e:
-            logger.error(f"Database error when fetching user by username '{username}': {e}", exc_info=True)
-            return None
-        finally:
-            if conn:
-                conn.close()
+    # Collaboration methods
+    def acquire_node_lock(self, script_id: str, node_id: str, user_id: str, session_id: str, duration_seconds: int = 300) -> bool:
```

Ruff (0.14.14)

[warning] Unused method argument: `script_id`

(ARG002)